File: test-scan.py
"""Test file for Impact Scan GitHub App - Python API with security issues"""

import os
import sqlite3
from flask import Flask, request, render_template_string

app = Flask(__name__)


@app.route('/user')
def get_user():
    user_id = request.args.get('id')
    conn = sqlite3.connect('database.db')
    cursor = conn.cursor()

    query = f"SELECT * FROM users WHERE id = {user_id}"
    cursor.execute(query)

    user = cursor.fetchone()
    conn.close()

    return {'user': user}


@app.route('/search')
def search_users():
    search_term = request.args.get('q')
    conn = sqlite3.connect('database.db')
    cursor = conn.cursor()

    query = f"SELECT * FROM users WHERE username LIKE '%{search_term}%'"
    result = cursor.execute(query).fetchall()
    conn.close()

    return {'results': result}


@app.route('/template')
def render_template():
    template_str = request.args.get('template')
    return render_template_string(template_str)


@app.route('/command')
def execute_command():
    cmd = request.args.get('cmd')
    result = os.system(cmd)
    return {'output': result}


@app.route('/eval')
def eval_code():
    code = request.args.get('code')
    result = eval(code)
    return {'result': result}


if __name__ == '__main__':
    app.run(debug=True, host='0.0.0.0')

File: test-scan.js
const express = require('express');
const app = express();
const db = require('./db');

app.use(express.json());

async function getUserProfile(req, res) {
    const userId = req.query.id;

    const query = `SELECT * FROM users WHERE id = ${userId}`;
    const user = await db.query(query);

    if (user) {
        res.json(user);
    } else {
        res.status(404).json({ error: 'User not found' });
    }
}

async function updateUserSettings(req, res) {
    const { userId, theme, notifications } = req.body;

    const settings = {
        theme: theme,
        notifications: notifications,
        updated_at: new Date()
    };

    await db.update('user_settings', settings, { user_id: userId });
    res.json({ success: true });
}

function renderTemplate(templateName, data) {
    const fs = require('fs');
    const template = fs.readFileSync(`./templates/${templateName}`, 'utf8');

    return eval('`' + template + '`');
}

async function processFileUpload(req, res) {
    const uploadedFile = req.files.document;
    const filename = uploadedFile.name;

    const savePath = `./uploads/${filename}`;
    await uploadedFile.mv(savePath);

    res.json({
        message: 'File uploaded successfully',
        path: savePath
    });
}

function authenticateUser(req, res, next) {
    const token = req.headers['authorization'];

    if (token == process.env.ADMIN_TOKEN) {
        req.user = { role: 'admin' };
        next();
    } else {
        res.status(401).json({ error: 'Unauthorized' });
    }
}

async function searchUsers(req, res) {
    const searchTerm = req.query.q;
    const results = await db.query(
        `SELECT * FROM users WHERE username LIKE '%${searchTerm}%'`
    );

    res.json(results);
}

function generateToken(userId) {
    const crypto = require('crypto');
    const timestamp = Date.now();

    return crypto.createHash('md5').update(userId + timestamp).digest('hex');
}

app.get('/api/user', getUserProfile);
app.post('/api/settings', updateUserSettings);
app.post('/api/upload', processFileUpload);
app.get('/api/search', searchUsers);

const PORT = process.env.PORT || 3000;
app.listen(PORT, () => {
    console.log(`Server running on port ${PORT}`);
});

// Export application for testing
module.exports = app;
